scipy/stats/_stats_py.py

```diff
@@ -1013,6 +1013,16 @@ def _moment(a, moment, axis, *, mean=None):
         # Starting point for exponentiation by squares
         mean = a.mean(axis, keepdims=True) if mean is None else mean
         a_zero_mean = a - mean
+
+        eps = np.finfo(a_zero_mean.dtype).resolution * 1e6
+        rel_diff = np.max(np.abs(a_zero_mean), axis=axis, keepdims=True)/mean
```

Comment:
@thomasjpfan thanks for your help today. This issue is related to that one (gh-10821) in that it also involves catastrophic cancellation when computing a moment from data.

In gh-10821, you looked for the variance being zero (`< np.finfo(dtype).tiny`) and did something to compensate. When this happened, you didn't really need to calculate the variance very accurately because this signaled that the optimization need not consider that value of `lmbda`.

This is general n-th order moment computation code, so there's not much we can do to compensate when precision loss occurs. So here, I'm looking for the relative difference between data and the mean being less than some tolerance, and I just want to warn the user that some sort of precision loss has occurred. I could probably be more specific about how much and exactly where it has occurred, but that can be left to a follow-up.

Do you have any thoughts on this PR?

Reply:
It's a trade off to me. Adding a precision loss check adds more computational, but it gives warnings when something looks wrong. My concern is the heuristic for checking the precision loss and making sure that does not warn for valid input. For example, how was `1e6` decided when computing `eps`?

As for the PR, if you want the bare mimumum change, one can directly detect the near constant case in `skew` and raise a specific warning for that.

Reply:
Thanks @thomasjpfan!  We already have an exact equality check in `skew` and `kurtosis`, but afterward we got a another report for which that check was insufficient. Also, we've gotten the same sort of issue in a few contexts, so I thought I'd treat more broadly. 
`1e6` was arbitrary, but it corresponds with a loss of approximately 10 decimal digits of precision for `float64`. This may be too noisy, so I'll reduce it so that this only warns when there is almost complete loss of precision.
Thanks!